components/oauth/org.wso2.carbon.identity.discovery/src/main/java/org/wso2/carbon/identity/discovery/internal/OIDCDiscoveryServiceComponent.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.discovery.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.carbon.identity.discovery.DefaultOIDCProcessor;
+import org.wso2.carbon.identity.discovery.OIDCProcessor;
+
+/**
+ * @scr.component name="identity.discovery.component" immediate="true"
+ */
+public class OIDCDiscoveryServiceComponent {
+    private static Log log = LogFactory.getLog(OIDCDiscoveryServiceComponent.class);
+    private static BundleContext bundleContext = null;
+
+    public static BundleContext getBundleContext() {
```

Comment:
This is a wrong way to get hold of bundleCtx instance. You should always use DataHolder pattern to store and retrieve object instances.